File: javascript/phone-number/phone-number.js
function getOnlyDigits(input) {
  return input.replace(/[^\d]/g, '');
}

var PhoneNumber = function(number){
  this.num = number;
};

PhoneNumber.prototype.number = function(){
  var parsedNumber = getOnlyDigits(this.num);
  if (parsedNumber.length === 10){
    return parsedNumber;
  }
  else if (parsedNumber.length === 11 && parsedNumber[0] === "1"){
    return parsedNumber.slice(1);
  }
  else {
    return '0000000000';
  }
};

PhoneNumber.prototype.areaCode = function(){
  return this.num.slice(0,3);
  };

PhoneNumber.prototype.toString = function(){
    return ['(',this.number().slice(0,3),')',' ',this.number().slice(3,6),'-',this.number().slice(6)].join('');
};

module.exports = PhoneNumber;

File: ruby/phone-number/phone_number.rb
require 'pry'
class PhoneNumber

  def initialize(str)
    @parsed_number = get_only_digits(str)
  end

  def number
    if @parsed_number.length == 10
      @parsed_number
    elsif @parsed_number.length == 11 && @parsed_number.slice(0) === "1"
       @parsed_number[1..-1]
    else
      '0000000000'
    end
  end

  def area_code
    @parsed_number[0..2]
  end

  def to_s
  "(" + number[0..2]+")" + " " + number[3..5] + "-" + number[6..-1]
  end

  def get_only_digits(str)
    return '0' * 10 if str.match(/[a-zA-Z]/)
    str.gsub(/[^\d]/, '')
  end
end
